### ISSUE TITLE
Cancel fulfilment endpoint missing after upgrade to 2022-07

### ISSUE THREAD

**Issue by thomotron** [nozzlegear/ShopifySharp#833]
As part of the upgrade to 2022-07, several deprecated fulfilment endpoints were removed (see af14afb8c5f5c4a3bfcbeec9cb7034906a9d4b1f) including `CancelAsync`. However, there was another cancellation endpoint available under [`/fulfillments/{fulfillment_id}/cancel.json`](https://shopify.dev/api/admin-rest/2022-07/resources/fulfillment#post-fulfillments-fulfillment-id-cancel) that hadn't been implemented and serves as a replacement to the old [`/orders/{order_id}/fulfillments/{fulfillment_id}/cancel.json`](https://shopify.dev/api/admin-rest/2022-04/resources/fulfillment#post-orders-order-id-fulfillments-fulfillment-id-cancel) endpoint which has since been removed.

As of ShopifySharp 5.19.0 there doesn't seem to be a way (from what I could find?) to cancel a fulfilment.

The old `CancelAsync` method could probably be reimplemented using the new endpoint and a slightly different signature to match the new parameters.

**Comment by nozzlegear** [comment]
Thanks! I don't often work with fulfillments, so just want to confirm: the `FulfillmentOrderService.CancelAsync` method does not do what you need it to do?

**Comment by thomotron** [comment]
Correct. The cancel endpoint for fulfilment orders is to cancel the fulfilment order as a whole rather than one single fulfilment.

The scenario I came across where this would be necessary is when cancelling a partially fulfiled order. If a fulfilment order has been closed successfully the API won't let you cancel it until you cancel the individual fulfilments assigned to the it. Once they're gone the fulfilment order becomes obsolete and the order can be cancelled.
